gateway/src/inference/providers/anthropic.rs

```diff
@@ -34,7 +33,7 @@ lazy_static! {
 }
 const ANTHROPIC_API_VERSION: &str = "2023-06-01";
 
-fn default_api_key_location() -> CredentialLocation {
+pub fn default_api_key_location() -> CredentialLocation {
```

Comment:
OK now that I see that you'll have to push these functions back outside I retract my previous statement about converting to a `Credential` prior to passing into the `new()` function. I think this becomes error prone if we have to rely on the developer getting the correct `default_api_key_location`. Can you make this private, change the interface back to `Option<CredentialLocation>` and use the `TryFrom` in each provider? We can also remove the conversion helper if we do this.